src/leaf.ml

```diff
@@ -37,7 +37,9 @@ functor
 
     let will_overflow t = LeafFmt.length t.leaf >= 2 * Params.fanout
 
-    let underflow t = LeafFmt.length t.leaf < Params.fanout
+    let underflow t =
+      (* I'm not sure [underflow] is the right naming here. *)
+      LeafFmt.length t.leaf < Params.fanout
```

Comment:
BTW, are you sure that this shouldn't be `LeafFmt.length t.leaf <= Params.fanout`.

Your 3 functions are (almost) exclusive. Maybe you should replace the 3 functions by just a `val state : t -> Underflow | Will_overflow | Overflow`

Reply:
I am sure :) and indeed it might make sense to do this refactoring, although those checks happen in very distinct situation (namely `insertion` and `remove`).